src/scenes/Event/components/EventTxHistory/index.js

```diff
@@ -76,31 +83,33 @@ export default class EventTxHistory extends Component {
   }
 
   renderTxRow = (transaction) => {
-    const { intl } = this.props;
+    const { intl, classes } = this.props;
     const { locale, messages: localeMessages } = intl;
+    const isExpanded = this.state.expanded.includes(transaction.txid);
 
     const result = [];
     result[0] = (
```

Comment:
ok, so here's a lesson in `<Fragment>`s and managing state for a list of items. Let's do this instead
```js
@withStyles(styles)
class TxRow extends Component {
  state = {
    expanded: false
  }

  toggle = () => this.setState({ expanded: !this.state.expanded })

  render() {
    const { transaction, classes, intl: { locale, localeMessages } } = this.props
    const { txid, createdTime, amount, token, type, status } = transaction
    const { expanded } = this.state

    return (
      <Fragment>
        <TableRow key={txid} onClick={this.toggle}>
          <TableCell padding="dense">{getShortLocalDateTimeString(createdTime)}</TableCell>
          <TableCell padding="dense">{getTxTypeString(type, locale, localeMessages)}</TableCell>
          <TableCell padding="dense">{description}</TableCell>
          <TableCell padding="dense">
            {!amount ? '' : `${amount} ${token}`}
          </TableCell>
          <TableCell padding="dense">{transaction.status}</TableCell>
          <TableCell padding="dense"><i className={cx(expanded ? 'icon-ic_down' : 'icon-ic_up', 'icon iconfont', classes.arrowSize)} /></TableCell>
        </TableRow>
        {expanded && (
          <Fragment>
            <TableRow selected onClick={this.toggle} key={`txaddr-${txid}`}>
              <TransactionHistoryID transaction={transaction} />
              <TableCell /><TableCell /><TableCell /><TableCell /><TableCell />
            </TableRow>
            <TableRow selected onClick={this.toggle} key={`txaddr-${txid}`}>
              <TransactionHistoryAddress transaction={props} />
              <TableCell /><TableCell /><TableCell /><TableCell /><TableCell />
            </TableRow>
          </Fragment>
        )}
      </Fragment>
    );
  }
}

```
and then in the render method
```
render() {
  return (
    ...
    <TableBody>
      {transactions.map((transaction, index) => (
        <TxRow {...transaction} description={this.getDescription(transaction)} key={transaction.txid} />
      ))}
    </TableBody>
    ...
  )
}
```

Reply:
we can cleanup the `getDescription` method later